src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -331,93 +382,76 @@ def handle_result(self, subscription: UptimeSubscription | None, result: CheckRe
+            # If the scheduled check time is older than the most recent update then we've already processed it.
+            # We can end up with duplicates due to Kafka replaying tuples, or due to the uptime checker processing
+            # the same check multiple times and sending duplicate results.
+            # We only ever want to process the first value related to each check, so we just skip and log here
+            metrics.incr(
+                "uptime.result_processor.skipping_already_processed_update",
+                tags={"mode": mode_name, **metric_tags},
+                sample_rate=1.0,
+            )
+            return
 
-            if project_subscription.mode == ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING:
-                self.handle_result_for_project_auto_onboarding_mode(
-                    project_subscription, result, metric_tags.copy()
-                )
-            elif project_subscription.mode in (
-                ProjectUptimeSubscriptionMode.AUTO_DETECTED_ACTIVE,
-                ProjectUptimeSubscriptionMode.MANUAL,
-            ):
-                self.handle_result_for_project_active_mode(
-                    project_subscription, result, metric_tags.copy()
-                )
-        except Exception:
-            logger.exception("Failed to process result for uptime project subscription")
+        if features.has("organizations:uptime-detailed-logging", organization):
+            logger.info("handle_result_for_project.after_dedupe", extra=result)
 
-        # Now that we've processed the result for this project subscription we track the last update date
-        cluster = _get_cluster()
+        if options.get("uptime.snuba_uptime_results.enabled"):
+            produce_snuba_uptime_result(project_subscription, result, metric_tags.copy())
+
+        # We log the result stats here after the duplicate check so that we
+        # know the "true" duration and delay of each check. Since during
+        # deploys we might have checks run from both the old/new checker
+        # deployments, there will be overlap of when things run. The new
+        # deployment will have artificially inflated delay stats, since it may
+        # duplicate checks that already ran on time on the old deployment, but
+        # will have run them later.
+        #
+        # Since we process all results for a given uptime monitor in order, we
+        # can guarantee that we get the earliest delay stat for each scheduled
+        # check for the monitor here, and so this stat will be a more accurate
+        # measurement of delay/duration.
+        record_check_metrics(result, project_subscription, {"mode": mode_name, **metric_tags})
+
+        # Track the last update date to allow deduplication
         cluster.set(
-            build_last_update_key(project_subscription),
+            last_update_key,
             int(result["scheduled_check_time_ms"]),
             ex=LAST_UPDATE_REDIS_TTL,
         )
```

Comment:
This also moved slightly earlier in the pipeline, though now that I'm looking at this maybe this really is wrong, since if there's a problem processing we probably want to allow the check to come in again and be processed again